crates/ibc/Cargo.toml

```diff
@@ -104,3 +105,4 @@ test-log = { version = "0.2.10", features = ["trace"] }
 tendermint-rpc = { version = "0.34", features = ["http-client", "websocket-client"] }
 tendermint-testgen = { version = "0.34" } # Needed for generating (synthetic) light blocks.
 parking_lot = { version = "0.12.1" }
+typed-builder = { version = "0.17.0" }
```

Comment:
@Farhad-Shabani should we use this to avoid redefining the dev dependencies?

https://github.com/rust-lang/cargo/issues/2911#issuecomment-749580481

With this, we can enable `ibc` crate with `mock` feature and drop the dev-deps for tendermint-testgen, typed-builder.

Reply:
Looks cool. Though not sure how much stable it is. I'll let you decide.

Reply:
I think the workaround is stable and part of the dependency resolver version 2 - which we are using already. But the ideal solution should be more direct.

I will add this, as this only affects dev-dependencies.

Reply:
Just one thing. This will enable `mocks` features by default in the test environment - which is equivalent to the current `cargo test --features mocks` command.

But IMO, this makes sense, as we want to run mock-based tests by default under `cargo test`.

Reply:
That's fine then.
Just take this into consideration as well: we will soon split out the mock module into a new crate :)